amplify/src/main/java/com/github/stkent/amplify/tracking/managers/BaseEventsManager.java

```diff
@@ -130,4 +141,8 @@ private T getCachedTrackingValue(@NonNull final IEvent event) {
         return value != null ? value : defaultTrackingValue();
     }
 
+    private boolean eventHasOccurredBefore(@NonNull final IEvent event) {
+        return !getCachedTrackingValue(event).equals(defaultTrackingValue());
```

Comment:
Now it's the responsibility of the manager to determine whether or not the event has occurred, which makes more sense since it's also the place where the default tracking value is specified (via subclassing)!